docs/SERVER-API.md

```diff
@@ -0,0 +1,214 @@
+
+
+## Basic controls
+
+### Write input
+
+**Purpose:** Writes a set of bytes to the terminal, not to be interpeted by zellij.
+
+**Parameters:** Vector of bytes to be written.
+
+**Response:** None.
+
+### Switch input mode
+@@@Is this part of the API or a client detail?
+
+**Purpose:** Switches between different zellij [input modes](./TERMINOLOGY.md#input_modes).
+
+**Parameters:** Which input mode to switch to.
+
+**Response:** None.
+
+### Scroll up
+
+**Purpose:** Scrolls up (back) in the screen buffer of the focus [pane](./TERMINOLOGY.md#pane).
+
+**Parameters:** None.
+
+**Response:** None.
+
+### Scroll down
+
+**Purpose:** Scrolls down (forwards) in the screen buffer of the focus [pane](./TERMINOLOGY.md#pane).
+
+**Parameters:** None.
+
+**Response:** None.
+
+### Terminal resize
+
+**Purpose:** Indicates that the terminal has been resized and so the layout should be redrawn.
+
+**Parameters:** New size of the layout.
+
+**Response:** @@@do we have a lock here?
+
+### Open file
+
+**Purpose:** Indicates that a new terminal should be spawned.
+
+**Parameters:** File name (optional) - the name of a file to be opened, using a terminal application determined by first the `EDITOR` and (if `EDITOR` is not set) `VISUAL` environment variables.
+
+**Response:** @@@do we have a lock here?
+
+## Pane controls
+
+### New pane
+
+**Purpose:** Opens a new [pane](./TERMINOLOGY.md#pane).
+
+**Parameters:** Direction (optional) - the position of the new pane relative to the focus pane.
+
+**Response:** Done opening new pane - indicates that the server has finished opening the pane. This exists for the purpose of synchronization @@@what's blocked?
+
+### Close pane
+
+**Purpose:** Close the focus [pane](./TERMINOLOGY.md#pane).
+
+**Parameters:** None.
+
+**Response:** Finished opening new pane - indicates that the server has finished closing the pane. This exists for the purpose of synchronization @@@what's blocked?
+
+### Switch focus pane
+
+**Purpose:** Switch the focus [pane](./TERMINOLOGY.md#pane). The focus pane is the target for many other pane-related actions.
+
+**Parameters:** @@@none?
+
+**Response:** None.
+
+### Move focus pane
+
+**Purpose:**
+
+**Parameters:** 
+
+**Response:** 
+
+### Toggle fullscreen focus pane
+
+**Purpose:** In a normal layout, switches the focus [pane](./TERMINOLOGY.md#pane) to be fullscreen. When the focus pane is fullscreen, switches back to the normal layout, potentially with multiple panes.
+
+**Parameters:** None.
+
+**Response:** Whether the layout has the focus pane locked to fullscreen or not.
+
+### Resize focus pane
+
+**Purpose:** Resizes the focus [pane](./TERMINOLOGY.md#pane). 
+
+**Parameters:** Direction in which the pane should be resized. @@@ When does it shrink, when does it grow? Is it the top left that's fixed?
+
+**Response:** None.
+
+### Horizontal split
+
+**Purpose:** Split the current focus [pane](./TERMINOLOGY.md#pane) horizontally, creating a new pane @@@above or below?
```

Comment:
On this and the next one, I don't know in what direction the new panes are spawned. @imsnif ?

Reply:
The new pane is spawned below in this case and towards the right in the next one.

Reply:
Yeah, in the UI we already changed this to the (IMO clearer) "split_right" / "split_down", might be a good idea to change it in the code-base and thus the API too :)